cores/esp8266/Updater.cpp

```diff
@@ -115,7 +117,11 @@ bool UpdaterClass::begin(size_t size, int command, int ledPin, uint8_t ledOn) {
 
   if (command == U_FLASH) {
     //address of the end of the space available for sketch and update
+#ifdef HOST_MOCK
+    uintptr_t updateEndAddress = (uintptr_t) -1;
+#else
     uintptr_t updateEndAddress = (uintptr_t)&_FS_start - 0x40200000;
+#endif
```

Comment:
Can this ifndef be avoided if something provides _FS_start symbol?

Reply:
Awesome idea.  Let me pull in your suggestion in the PR that fixes the uninitted write() buffer valgrind error.